source/simulator/core.cc

```diff
@@ -686,6 +686,31 @@ namespace aspect
     // wait if there is a thread that's still writing the statistics
     // object (set from the output_statistics() function)
     output_statistics_thread.join();
+
+    // Detect if we are being destructed because an uncaught exception has been triggered.
+    if (std::uncaught_exception())
+      {
+#ifdef DEAL_II_WITH_CXX11
+        std::cout << "ERROR: uncaught exception detected:" << std::endl;
+        try
+          {
+            auto eptr = std::current_exception();
+            if (eptr)
+              std::rethrow_exception(eptr);
+          }
+        catch (const std::exception &e)
+          {
+            std::cout << e.what() << std::endl;
+          }
+        throw; // terminate to avoid deadlock in ~TimerOutput
```

Comment:
Hm, you are throwing an exception in a destructor. This isn't allowed.

Also, which exception is this exactly? (I think I don't understand C++ well enough :-) )

Reply:
Yes. I want to terminate.